### PR TITLE
Improvements for waiters + more resilience for fill-form

### PR DESCRIPTION
**Pull request by tlipski**
Main changes:
1. Rework waiters to always throw TimeoutException upon exceeding the timeout. The exception contains information about the checker function.
2. Rework fill form to wait for the controls to become available and re-try in case of error, making it work with complex AngularJS apps better.

### CODE REVIEW

File: src/io/aviso/taxi_toolkit/waiters.clj [hunk 1]
```diff
@@ -7,18 +7,33 @@
             [io.aviso.taxi-toolkit
              [ui :refer :all]
              [utils :refer :all]
-             [assertions :as at]]))
+             [assertions :as at]])
+  (:import (java.util.regex Pattern)))
+
+(defn smart-wait [checker el-spec]
```
Comment (kamituel) [inline src/io/aviso/taxi_toolkit/waiters.clj]: Could the arglist be in a new line?

File: src/io/aviso/taxi_toolkit/waiters.clj [hunk 2]
```diff
@@ -52,17 +75,15 @@
   the given pattern or string."
   [url]
   (try
-    (wait-until #(re-find (re-pattern url) (current-url)))
-    (catch org.openqa.selenium.TimeoutException err
-      (is false (str "Expected URL to match: '" url "' but got: '" (current-url)"'")))))
+    (wait-until #(re-find (re-pattern url) (current-url)) webdriver-timeout)
+    (catch org.openqa.selenium.TimeoutException e
+      (throw (org.openqa.selenium.TimeoutException. (str "Timeout exceeded for: " (current-url) " to match: " url) e))
+      )))
```
Comment (kamituel) [inline src/io/aviso/taxi_toolkit/waiters.clj]: Should be at the and of prev line.

**Comment by kamituel** [comment]
Besides minor comments, looks good :)

File: src/io/aviso/taxi_toolkit/waiters.clj (hunk 1 above)
Comment (tlipski) [inline src/io/aviso/taxi_toolkit/waiters.clj]: of course

File: src/io/aviso/taxi_toolkit/waiters.clj (hunk 2 above)
Comment (tlipski) [inline src/io/aviso/taxi_toolkit/waiters.clj]: no problem at all

**Comment by tlipski** [comment]
@kamituel please have a look at the 3946e9f commit

**Comment by tlipski** [comment]
Thanks!

**Pull request merged**